app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationSample.java

```diff
@@ -1,31 +1,208 @@
 package com.teamagam.gimelgimel.app.model.entities;
 
+import android.location.Location;
+import android.location.LocationManager;
+
 import com.google.gson.annotations.SerializedName;
 import com.teamagam.gimelgimel.app.view.viewer.data.geometries.PointGeometry;
 
-import java.util.Date;
-
 /**
- * Created by Yoni on 4/18/2016.
+ * Created on 4/18/2016.
+ * <p/>
+ * An immutable data class representing a geographic location sample.
+ * <p/>
+ * <p>A location can consist of a {@link PointGeometry} , timestamp,
+ * and other information such as bearing, altitude and velocity.
+ * <p/>
+ * <p>All locations generated by the {@link LocationManager} are
+ * guaranteed to have a valid latitude, longitude, and timestamp
+ * (UTC time), all other
+ * parameters are optional.
  */
 public class LocationSample {
 
-//    --------MOCKING-------
-
     @SerializedName("location")
     private PointGeometry mPoint;
 
     @SerializedName("timeStamp")
-    private Date mTime;
+    private long mTime;
+
+    @SerializedName("provider")
+    private String mProvider;
+
+    @SerializedName("hasSpeed")
+    private boolean mHasSpeed = false;
+
+    @SerializedName("speed")
+    private float mSpeed = 0.0f;
+
+    @SerializedName("hasBearing")
+    private boolean mHasBearing = false;
+
+    @SerializedName("bearing")
+    private float mBearing = 0.0f;
 
-    @SerializedName("userName")
-    private String mUserId;
+    @SerializedName("hasAccuracy")
+    private boolean mHasAccuracy = false;
 
-    public LocationSample(PointGeometry point, Date time, String userId){
+    @SerializedName("accuracy")
+    private float mAccuracy = 0.0f;
+
+    /**
+     * Construct a new Location Sample that has only time and location.
+     */
+    public LocationSample(PointGeometry point, long time) {
         mPoint = point;
         mTime = time;
-        mUserId = userId;
     }
 
-    //get + set
+    /**
+     * Construct a new Location Sample that is copied from an existing Location.
+     */
+    public LocationSample(Location l) {
```

Comment:
Usually, constructor doesn't implement any logic/parsing.
When the need arises, a factory method can be used.